app/src/main/java/org/simple/clinic/navigation/v2/ScreenResult.kt

```diff
@@ -0,0 +1,13 @@
+
+package org.simple.clinic.navigation.v2
+
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+sealed class ScreenResult: Parcelable
+
+@Parcelize
+data class Succeeded(val result: Parcelable): ScreenResult()
+
+@Parcelize
+data class Failed(val cause: Parcelable? = null): ScreenResult()
```

Comment:
Ah, yes. It's so that we can return additional information to the calling class if we want to handle different failure cases. If the result failure is some kind of an exception, say, we can process that exception in the class which is supposed to do the exception and then provide typed values (like a `sealed class FailureReason`) to the calling class for safer error handling.